src/utils/records.ts

```diff
@@ -88,17 +88,21 @@ export class Records {
   /**
    * Constructs the full key derivation path using `dataFormats` scheme.
    */
-  private static constructKeyDerivationPathUsingDataFormatsScheme(
-    recordId: string,
+  public static constructKeyDerivationPathUsingDataFormatsScheme(
     descriptor: RecordsWriteDescriptor
   ): string[] {
-    const fullDerivationPath = [
-      KeyDerivationScheme.DataFormats,
-      descriptor.dataFormat,
-      recordId
-    ];
-
-    return fullDerivationPath;
+    if (descriptor.schema !== undefined) {
+      return [
+        KeyDerivationScheme.DataFormats,
+        descriptor.schema, // this is as spec-ed on TP27, the intent is to support sharing the key for just a specific data type under a schema
+        descriptor.dataFormat
```

Comment:
@thehenrytsai I remember we talked about the fact that the TP27 seems to spec this, but I'm still not convinced that this is or isn't the right thing to do. What was the reason/use case why we don't want to share a key for all records of a given `dataFormat`, including records with different `schemas`?

Reply:
I extrapolate the reason being the same as why spec does not support sharing a key for all records of a given schema, including protocol-space records.

Reply:
\I definitely agree with sharply dividing access of non-protocol schema from protocol records. So, tentatively, following that pattern for dataFormat works for me.